shanoir-ng-front/src/app/dataset-acquisitions/dataset-acquisition/dataset-acquisition.component.ts

```diff
@@ -125,4 +128,18 @@ export class DatasetAcquisitionComponent extends EntityComponent<DatasetAcquisit
     downloadAll() {
         this.downloadService.downloadAllByAcquisitionId(this.datasetAcquisition?.id, this.downloadState);
     }
+
+    getCopiedEntity(copyMsg: string) {
```

Comment:
If the message field is intended to retrieve ids by parsing it here, it's not the right way to do it. The copied entity should have a copiedFrom attribute/column that references the original entity. Note that if this reference should remain when the origin entity is deleted, then it cannot have a foreign key constraint. Then the message would be reconstructed by the front using this id.

Reply:
Every (examination, acquisition, datasets) entity has a "source" value. If it's not empty, it means it's a copy and the id stored in 'source' is the id of the original entity. Otherwise, it's null.
We can't delete an original entity if a copy of it exists